histomicstk/utils/Sample.py

```diff
@@ -16,116 +15,81 @@ def Sample(File, Magnification, Percent, Tile, MappingMag=1.25, Coverage=0.1):
-    LR = np.asarray(LR)
-    LR = LR[:, :, :3]
-
-    # resize if desired magnification is not provided by the file
-    if lrSchedule.Factor != 1.0:
-        LR = scipy.misc.imresize(LR, lrSchedule.Factor)
-        lrHeight = LR.shape[0]
-        lrWidth = LR.shape[1]
-
-    # mask
-    Mask = SimpleMask(LR)
-
-    # pad mask to match overall size of evenly tiled image
-    MaskHeight = lrSchedule.Tout * lrSchedule.Y.shape[0]
-    MaskWidth = lrSchedule.Tout * lrSchedule.X.shape[1]
-    LRMask = np.zeros((MaskHeight, MaskWidth), dtype=np.uint8)
-    LRMask[0:lrHeight, 0:lrWidth] = Mask
-
-    # sample from tile at full resolution that contain more than 1/2 foreground
-    Pixels = list()
-    for i in range(Schedule.X.shape[0]):
-        for j in range(Schedule.X.shape[1]):
-            lrTileMask = LRMask[
-                i * lrSchedule.Tout:(i + 1) * lrSchedule.Tout,
-                j * lrSchedule.Tout:(j + 1) * lrSchedule.Tout].astype(np.uint8)
-            TissueCount = sum(lrTileMask.flatten().astype(
-                np.float)) / (lrSchedule.Tout**2)
-            if TissueCount > Coverage:
-                # region from desired magnfication
-                Tile = Slide.read_region((int(Schedule.X[i, j]),
-                                          int(Schedule.Y[i, j])),
-                                         Schedule.Level,
-                                         (Schedule.Tout, Schedule.Tout))
-                Tile = np.asarray(Tile)
-                Tile = Tile[:, :, :3]
-
-                # resize if desired magnification is not provided by the file
-                if Schedule.Factor != 1.0:
-                    Tile = scipy.misc.imresize(Tile,
-                                               Schedule.Factor,
-                                               interp='nearest')
-
-                # upsample tile mask from low-resolution to high-resolution
-                TileMask = scipy.misc.imresize(lrTileMask,
-                                               Tile.shape,
-                                               interp='nearest')
-
-                # generate linear indices of pixels in mask
-                Indices = np.nonzero(TileMask.flatten())[0]
-                Sampling = np.random.uniform(0, Indices.size,
-                                             (np.ceil(Percent *
-                                                      Indices.size),))
-                Indices = Indices[Sampling.astype(np.int)]
-
-                # convert rgb tile to 3xN array and sample with linear indices
-                Vectorized = np.reshape(Tile,
-                                        (Tile.shape[0] * Tile.shape[1], 3))
-                Pixels.append(Vectorized[Indices, :].transpose())
+    ts = large_image.getTileSource(slide_path)
+
+    # get enitre whole-silde image at low resolution
+    im_lowres = ts.getRegion(scale={'magnification': mapping_mag},
+                             format=large_image.tilesource.TILE_FORMAT_NUMPY)
```

Comment:
The `tileIterator` can return a numpy array, but we haven't added that option to `getRegion`.  I can (please add an issue and I'll get to it).